subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheTaskExecutionIntegrationTest.groovy

```diff
@@ -16,7 +16,33 @@
 
 package org.gradle.configurationcache
 
+import spock.lang.Issue
+
 class ConfigurationCacheTaskExecutionIntegrationTest extends AbstractConfigurationCacheIntegrationTest {
+
+    @Issue('https://github.com/gradle/gradle/issues/22522')
+    def "reports problem for extra property accessed at execution time"() {
+        given:
+        buildKotlinFile '''
+            tasks.register("report") {
+                extra["outputFile"] = file("$buildDir/output.txt")
+                outputs.files(extra["outputFile"])
+                doLast {
+                    val outputFile: File by extra
+                    outputFile.writeText("")
+                }
+            }
+        '''
+
+        when:
+        configurationCacheFails(WARN_PROBLEMS_CLI_OPT, 'report')
```

Comment:
@bamboo It seems like because load-after-store we can't expect first run being successful. Am I right?

Reply:
Right!